pymc3/distributions/distribution.py

```diff
@@ -458,39 +490,55 @@ def random(self, point=None, size=None, repeat=None):
                                 size=size)
 
     def logp(self, value):
-        return bound(self.dist.logp(value),
-                     value >= self.lower, value <= self.upper)
+        logp = self._wrapped.logp(value)
+        bounds = []
+        if self.lower is not None:
+            bounds.append(value > self.lower)
+        if self.upper is not None:
+            bounds.append(value < self.upper)
+        if len(bounds) > 0:
+            return bound(logp, *bounds)
+        else:
+            return logp
 
 
 class Bound(object):
     R"""
-    Creates a new upper, lower or upper+lower bounded distribution
+    Create a new upper, lower or upper+lower bounded distribution.
+
+    The resulting distribution is not normalized anymore. This
+    is usually fine if the bounds are constants. If you need
+    truncated distributions, use `Bound` in combination with
+    a `pm.Potential` with the cumulative probability function.
```

Comment:
We probably want an example of this. Maybe just in the `examples` folder, rather than a full-blown notebook.